TOZ_iOS/ArrayResponseMapper.swift

```
@@ -7,15 +7,17 @@
 
 import Foundation
 
-final class ArrayResponseMapper<A: ParsedItem> {
+class ArrayResponseMapper<A: ParsedItem> {
 
-    static func process(_ obj: AnyObject?, mapper: ((Any?) throws -> A)) throws -> [A] {
+    static func process(_ obj: AnyObject?, mapper: ((AnyObject?) throws -> A?)) throws -> [A?] {
         guard let json = obj as? [[String: AnyObject]] else { throw ResponseMapperError.invalid }
 
         var items = [A]()
         for jsonNode in json {
-            let item = try mapper(jsonNode)
-            items.append(item)
+            let item = try mapper(jsonNode as AnyObject)
+            if let item = item {
+                items.append(item)
+            }
```

We should throw an error if one of the elements parsing failed because currently we wouldn't know if something went wrong.